commitment/taproot.go

```diff
@@ -99,6 +99,34 @@ func NewPreimageFromLeaf(leaf txscript.TapLeaf) *TapscriptPreimage {
 	}
 }
 
+// ScriptFromLeafPreimage decodes the leafVersion and script from a taproot
+// leaf preimage.
+func ScriptFromLeafPreimage(preimage []byte) (*txscript.TapscriptLeafVersion,
```

Comment:
Re the comment below, we can have this accept `TapscriptLeaf`, then this becomes a transformation that extracts the versions, and returns that along with the encoded script.